test: Fix deprecation warning on incorrect assertion

Ferenc- opened the pull request:
Fixes the following error message
````
DEPRECATED: Use assert_nil if expecting nil
from /workspace/ruby-sensor/test/secrets_test.rb:97.
This will fail in Minitest 6.
````

GSVarsha reviewed (changes requested):
I think we should consider using `assert_nil` in the below lines as well. We've already changed [here](https://github.com/instana/ruby-sensor/commit/cd883d74276db0a023eb08b1183b9fed636b473e).

[tracer_test.rb#L124](https://github.com/instana/ruby-sensor/blob/fix-deprecated-warning/test/tracing/tracer_test.rb#L124)
[tracer_test.rb#L154](https://github.com/instana/ruby-sensor/blob/fix-deprecated-warning/test/tracing/tracer_test.rb#L154)

Ferenc- commented:
> I think we should consider using `assert_nil` in the below lines as well. We've already changed it [here](https://github.com/instana/ruby-sensor/commit/cd883d74276db0a023eb08b1183b9fed636b473e) in a previous commit.
> 
> [tracer_test.rb#L124](https://github.com/instana/ruby-sensor/blob/fix-deprecated-warning/test/tracing/tracer_test.rb#L124) [tracer_test.rb#L154](https://github.com/instana/ruby-sensor/blob/fix-deprecated-warning/test/tracing/tracer_test.rb#L154)

Good point. I have changed these and some more in sidekiq tests.

GSVarsha reviewed (approved):
Looks perfect to me!

arjun-rajappa reviewed (approved):
Looks good!
